File: v2/lang/ruby/vtry.rb
#! /usr/bin/env ruby
# vtry.rb -=- Try Vlerq from Ruby

require 'thrive'
w = Thrive::Workspace.new
p w
w.load(' 0 exit exec -1 syst [[ ')
p w.find('+')
p w.eval('54321')
p w.run
p w.pop
w.push(123,456)
p w.eval('+')
p w.run
p w.pop
w.push(12,34)
p w.run(w.find('+'))
p w.pop
w.push([1,nil,'a'])
p w.pop

__END__
Output:

#<Thrive::Workspace:0xb7d3e278>
#<Thrive::Reference:0xb7d3e228>
149
0
54321
33
0
579
-1
46
[1, nil, "a"]

File: v2/lang/lua/vtry.lua
#! /usr/bin/env lua

require 'thrive'
w = thrive.new()
print(w)
w:load(' 0 exit exec -1 syst [[ ')
print(w:find('+'))
print(w:eval('54321'))
print(w:run())
print(w:pop())
w:push(123,456)
print(w:eval('+'))
print(w:run())
print(w:pop())
w:push(12,34)
print(w:run(w:find('+')))
print(w:pop())
w:push({1,nil,'a'})
x = w:pop()
for i,j in pairs(x) do
  print(i,j)
end
print(#x)

--[[ output:

thrive_ws (0x8070c68)
thrive_ref (1 @ 0x8070c68)
149
0
54321
33
0
579
-1
46
1       1
3       a
3

--]]
